### ISSUE TITLE
multer don't show a file uploaded

### ISSUE THREAD

**Issue by Douglasgomestosta** [oven-sh/bun#2026]
### What version of Bun is running?

0.5.5

### What platform is your computer?

Linux 5.4.0-139-generic x86_64 x86_64

### What steps can reproduce the bug?

when I use "Multer" to send files to the server, the file simply does not appear, and the variable req.file returns undefined

`app.post('/testeee', parser.single('file'), function (req, res, next) {
    // req.file is the `avatar` file
    // req.body will hold the text fields, if there were any
    console.log(req.file);
  })`

the script does not display any error message

multer: https://www.npmjs.com/package/multer

### What is the expected behavior?

when running the same code in nodejs, the results are:
{
  fieldname: 'file',
  originalname: 'teste.jpeg',
  encoding: '7bit',
  mimetype: 'image/jpeg',
  destination: 'arquivos_temporarios',
  filename: '8e35b4be4885d8f660b29a5d9faad0af',
  path: 'arquivos_temporarios/8e35b4be4885d8f660b29a5d9faad0af',
  size: 138319
}

### What do you see instead?

undefined

### Additional information

i'm using expressjs

**Comment by sp90** [comment]
This works for me in bun v0.6.13

Pretty sure this fixed it

https://github.com/oven-sh/bun/pull/3467

**Comment by BitByBen** [comment]
Yeah. multer isn't even running for me. Server just sits there holding the request. Never hits my request handler code. 

Using Bun 1.0.2 on Ubuntu. 

```typescript 
// imports 
const upload = multer();
// ...
app.post(
  "/api/analysis",
  upload.fields([
    { name: "spec", maxCount: 1 },
    { name: "proposal", maxCount: 1 },
  ]),
  async (req: any, res: express.Response) => {
    // never runs
  });
```

**Comment by sp90** [comment]
@benjaminjs you're welcome to make an example repo and i can investigate

**Comment by NimishKashyap** [comment]
@sp90 
Can you please loook into it. I'm also facing the same isssue with bun. Need a quick fix as I need to solve it within 2 days.

https://github.com/LegalEase-SIH/backend/blob/features/nimish/Backend/controllers/petition.controller.ts

**Comment by sp90** [comment]
@NimishKashyap cant access your private repo so you need to make a small reproducable example

**Comment by NimishKashyap** [comment]
@sp90 Extremely sorry for the broken link. Here is the gist.

https://gist.github.com/NimishKashyap/88b4a44672fafdd120a597957110a49c

**Comment by swagisays** [comment]
is their any solution?

**Comment by wfzyx** [comment]
I can confirm that this still happens on 1.0.3

I'm attaching my minimal test case.

It needs `bun add express multer`

FWIW, the server is accepting the file and saving it in the file system, but multer never gives control back to express, thus the callback is never called, hanging until timeout.

```javascript
const express = require('express');
const multer = require('multer');
const storage = multer.diskStorage({
  destination: (req, file, callBack) => {
    callBack(null, 'public/images/');
  },
  filename: (req, file, callBack) => {
    callBack(
      null,
      file.originalname.split('.')[0] + '-' + Date.now() + path.extname(file.originalname)
    );
  }
});

const upload = multer({
  storage: storage
});

app.post('/done', upload.array('multiFilesDone', 50), callback);
const serverOptions = {
  hostname: "localhost",
  port: "5000"
};

app.listen(serverOptions, () => {
  console.log(`Server is listening at http://${serverOptions.hostname}:${serverOptions.port}`);
});
```

**Comment by swagisays** [comment]
any solution?

**Comment by asta-nguyen** [comment]
I am encountering a similar problem with `bun` version `1.0.6`.

**Comment by larsw** [comment]
Still a problem in bun 1.0.13 AFAIK.

**Comment by aditya-kumarr** [comment]
I faced this issue with bun 1.02, just updated it to 1.0.19 (latest) and it's working fine.

Update: I'm facing similar issue with Multer s3 when trying to use AUTO_CONTENT_TYPE

**Comment by aditya-kumarr** [comment]
Update, I'm facing similar issue with Multer s3 when trying to user AUTO_CONTENT_TYPE

**Comment by sp90** [comment]
@paperdave based on @aditya-kumarr comment -  I think this issue is closeable should i try to identify the AUTO_CONTENT_TYPE issue here or open a new issue and closing this?

**Comment by sp90** [comment]
I just updated this project using bun `1.0.21+fa7e4bce2` (current canary) and I both get the file data and the file is written to the uploads folder - this is as barebones as it goes using bun, express & multer

<img width="425" alt="Screenshot 2024-01-05 at 13 12 13" src="https://github.com/oven-sh/bun/assets/3041602/a0f29882-0e99-401a-b63b-0ac5668c4590">

Repo:
https://github.com/sp90/bun-multer-busboy

**Comment by paperclover** [comment]
> @paperdave based on @aditya-kumarr comment - I think this issue is closeable should i try to identify the AUTO_CONTENT_TYPE issue here or open a new issue and closing this?

lets use this issue to track current bugs with multer and hopefully we can get them all fixed. i havent looked too far into this just now but it seems theres some bug so i'll keep it open.

**Comment by sp90** [comment]
I'll investigate further into what bugs there are related to multer-s3 and post Them here in the coming week

**Comment by sp90** [comment]
@paperdave all these work as expected i have updated my repo to make it testable

https://github.com/sp90/bun-multer-busboy

I believe this ticket should be closed

**Comment by airamstudio** [comment]
Still happens to me specially after there's a session set with express-session. When I remove the session Multer starts working again (although still hangs sometimes but much less)
